src/Navbar/Nav.tsx

```diff
@@ -26,8 +27,19 @@ export default function Nav() {
             console.log("user is not logged in or an error has occured", err)
             evauluateLogin()
         })
+
+        
     }, [])
 
+    useEffect(() => {
```

Comment:
So this is going to run every single time the user goes to a new page. A work around is to save userData in the browser's session rather than the state. So remove the db call if window.sessionStorage.get('user') else make the call and set window.sessionStorage.set('user', userData). Make sure you set the state in both calls to prevent hook state bugs